gateway/ln-gateway/src/utils.rs

```diff
@@ -3,34 +3,41 @@ use std::{future::Future, result::Result, time::Duration};
 use tokio::time::sleep;
 use tracing::info;
 
-/// Retry an operation util the operation succeeds, OR
-/// The maximum number of attempts are made without success
-pub async fn retry<F, R, T>(
+/// Run the given closure up to `max_attempts` times while waiting for
+/// the specified `Duration` between attempts
+///
+/// # Returns
+///
+/// - If the closure is run succesfully, the result is immediately
+///   returned.
+/// - The closure is run at most `max_attempts` times after which the
+///   error of the closure is returned.
+pub async fn retry<F, Fut, T>(
     op_name: String,
     op_fn: F,
     wait: Duration,
-    max_retries: u32,
+    max_attempts: u32,
 ) -> Result<T, anyhow::Error>
 where
-    F: Fn() -> R,
-    R: Future<Output = Result<T, anyhow::Error>>,
+    F: Fn() -> Fut,
+    Fut: Future<Output = Result<T, anyhow::Error>>,
 {
-    let mut att = 0;
+    let mut counter = 1;
     loop {
         match op_fn().await {
             Ok(result) => return Ok(result),
-            Err(e) => {
-                att += 1;
-                if att > max_retries {
-                    return Err(e);
+            Err(err) => {
+                if counter == max_attempts {
```

Comment:
This introduces a new bug/feature if `max_attempts == 0`: the code will run forever.

Reply:
yeah, good point. at least it'll run `u32::MAX` times until it overflows in release mode